windows/test-app.js

```diff
@@ -1,18 +1,37 @@
+#!/usr/bin/env node
+
 //@ts-check
 const path = require("path");
 const fs = require("fs");
 
-const windowsDir = "windows";
+const { argv } = require("yargs")
+  .usage("Usage: $0 --projectDirectory=directory --applicationName=name")
```

Comment:
Can we read `applicationName` from `app.json`?

Reply:
The thought behind this was to use app name provided by user when running plop (by running `test-app.js` with correct parameters from `test-app.bat`). But that wouldn't work if we have `bin` script which runs `test-app.js` directly. So sure, we can use `applicationName` from the manifest.

Reply:
Would this mean that if `app.json` doesn't exist, we don't create solution at all?

Reply:
Hmm… Can we output a warning and fall back to "ReactTestApp.sln"?